roles/core/nic/molecule/default/verify.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Verify
+  hosts: all
+
+  tasks:
+  - name: check all files have been generated
+    find:
+      paths: /etc/sysconfig/network-scripts
+      patterns: "ifcfg-eth*"
+    register: ifcfg_list
+
+  - name: assert we have 3 eth files
+    assert:
+      that: ifcfg_list.matched == 3
+
+  - name: assert eth0
+    lineinfile:
+      path: /etc/sysconfig/network-scripts/ifcfg-eth0
+      line: "IPADDR={{ hostvars[inventory_hostname]['ansible_eth0'].ipv4.address }}"
```

Comment:
Can you double check the test does what you want? Does the verify fail when there is a change?

```
[root@management1 ~]# cat /etc/ansible//playbooks/test.yml
- name: Verify
  hosts: localhost

  tasks:
    - name: assert
      lineinfile:
        path: /tmp/ifcfg-eth0
        line: "IPADDR=1.2.3.4"
    - name: assert
      lineinfile:
        path: /tmp/ifcfg-eth1
        line: "IPADDR=4.3.2.1"
```
Create first file with expected IP address:
`[root@management1 ~]# echo IPADDR=1.2.3.4 > /tmp/ifcfg-eth0`

Create second file with wrong IP address:
`[root@management1 ~]# echo IPADDR=4.3.1.2 > /tmp/ifcfg-eth1`

Run the playbook:
```
[root@management1 ~]# ansible-playbook /etc/ansible//playbooks/test.yml

PLAY [Verify] **************************

TASK [Gathering Facts] *****************
ok: [localhost]

TASK [assert] **************************
ok: [localhost]

TASK [assert] **************************
changed: [localhost]

PLAY RECAP *****************************
localhost : ok=3 changed=1 unreachable=0 failed=0 skipped=0 rescued=0 ignored=0
```

Check the files:
```
[root@management1 ~]# cat /tmp/ifcfg-eth0
IPADDR=1.2.3.4
[root@management1 ~]# cat /tmp/ifcfg-eth1
IPADDR=4.3.1.2
IPADDR=4.3.2.1
```

Reply:
Given the doc, it should, but I will double check, since now I have a doubt!

Reply:
OK. Depending on the result of your test, can you add a comment before the lineinfile to specify the test should fail if the task report a change?